components/mmu_server.py

```diff
@@ -105,100 +669,133 @@ def gcode_processed_already(file_path):
-
-            match = tools_regex.match(line)
-            if match:
-                tool = match.group("tool")
-                tools_used.add(int(tool))
-
-            # !colors! processing
-            if not has_colors_placeholder and METADATA_COLORS in line:
-                has_colors_placeholder = True
-
-            if not found_colors:
-                match = colors_regex.match(line)
-                if match:
-                    colors_csv = [color.strip().lstrip('#') for color in match.group(1).split(';')]
-                    if not colors:
-                        colors.extend(colors_csv)
-                    else:
-                        colors = [n if o == '' else o for o,n in zip(colors,colors_csv)]
-                    found_colors = all(len(c) > 0 for c in colors)
-
-            # !temperatures! processing
-            if not has_temps_placeholder and METADATA_TEMPS in line:
-                has_temps_placeholder = True
-
-            if not found_temps:
-                match = temps_regex.match(line)
-                if match:
-                    temps_csv = re.split(';|,', match.group(2).strip())
-                    temps.extend(temps_csv)
-                    found_temps = True
-
-            # !materials! processing
-            if not has_materials_placeholder and METADATA_MATERIALS in line:
-                has_materials_placeholder = True
-
-            if not found_materials:
-                match = materials_regex.match(line)
-                if match:
-                    materials_csv = match.group(1).strip().split(';')
-                    materials.extend(materials_csv)
-                    found_materials = True
-            
-            # !purge_volumes! processing
-            if not has_purge_volumes_placeholder and METADATA_PURGE_VOLUMES in line:
-                has_purge_volumes_placeholder = True
-            
-            if not found_purge_volumes:
-                match = purge_volumes_regex.match(line)
+                    slicer = match.group(1)
+    if slicer in AUTHORZIED_SLICERS:
+        if isinstance(TOOL_DISCOVERY_REGEX, dict):
+            tools_regex = re.compile(TOOL_DISCOVERY_REGEX[slicer], re.IGNORECASE)
+        else:
+            tools_regex = re.compile(TOOL_DISCOVERY_REGEX, re.IGNORECASE)
+        if isinstance(COLORS_REGEX, dict):
+            colors_regex = re.compile(COLORS_REGEX[slicer], re.IGNORECASE)
+        else:
+            colors_regex = re.compile(COLORS_REGEX, re.IGNORECASE)
+        if isinstance(TEMPS_REGEX, dict):
+            temps_regex = re.compile(TEMPS_REGEX[slicer], re.IGNORECASE)
+        else:
+            temps_regex = re.compile(TEMPS_REGEX, re.IGNORECASE)
+        if isinstance(MATERIALS_REGEX, dict):
+            materials_regex = re.compile(MATERIALS_REGEX[slicer], re.IGNORECASE)
+        else:
+            materials_regex = re.compile(MATERIALS_REGEX, re.IGNORECASE)
+        if isinstance(PURGE_VOLUMES_REGEX, dict):
```

Comment:
Isn't that redundant, as the regexes are declared above and can't be dicts?

Reply:
I'd probably remove that, add a comment instead like "Do it as above if so", as having less complexity is always better! But as I'm not the owner, @moggieuk can decide, that's only if this project was mine.